pkg/controller/storagecluster/uninstall_reconciler.go

```diff
@@ -406,8 +384,8 @@ func (r *ReconcileStorageCluster) deleteResources(sc *ocsv1.StorageCluster, reqL
 	return nil
 }
 
-// deleteCephFilesystems deletes the CephFilesystems owned by the StorageCluster
-func (r *ReconcileStorageCluster) deleteCephFilesystems(sc *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+// ensureDeleted deletes the CephFilesystems owned by the StorageCluster
```

Comment:
I do not agree with anything  below this line.

Since we're adding an Interface and concrete types which implements it, I believe the concrete type and all it's method that it requires to implement the interface should be in the same package.

i.e. If we want `ocsCephFilesystems` to implement `resourceManager` interface, we should have `ensureCreated` and `ensureDeleted` under `storagecluster/cephfilesystem.go` instead of putting it in separte places.

This makes it readable, understandable and traceable. With current approach I'd have to jump through the code-base to understand why a certain concrete type does or does not implement an interface.

All that `storagecluster/uninstall_reconciler.go` needs to do is call the `ensureDeleted` and not implement the method for them.

Reply:
Hi @umangachapagain You are right We do have a plan for this in separate PR's

Reply:
Is it that big of a task to wait for next PR(s)?
Is this PR immediately required?

If not I'd like to make the right design changes in this PR.

Reply:
Done